File: 6. ZigZag Conversion.java
public class Solution {
    public String convert(String s, int numRows) {
        List<String> list = new ArrayList<String>();
        
        Boolean dir = true;
        int row = 0;
        
        for(int i = 0; i < numRows; i++) {
            list.add("");
        }
        
        for(int i = 0; i < s.length(); i++) {
            list.set(row%numRows, list.get(row%numRows) + (s.charAt(i) + ""));
            
            if(row%numRows == 0 && !dir) {
                dir = true;
                row = 1;
            } else if (row%numRows == numRows - 1 && dir) {
                row--;
                dir = false;
            } else if (dir) {
                row++;
            } else {
                row--;
            }
        }
		
		String s1 = "";
        
        for(int i = 0; i < numRows; i++) {
            s1 += list.get(i);
        }
        return s1;
    }
}

File: 6. ZigZag Conversion.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        vector<string> ans(numRows); 
        bool dir = true;
        string::iterator iter = s.begin();
        
        while(iter < s.end()) {
            for(int i = 0; i < numRows && iter < s.end(); i++, iter++) {
                ans.at(i) = ans.at(i) + *iter; 
            }
            
            for(int i = numRows - 2; i > 0 && iter < s.end(); i--, iter++) {
                ans.at(i) = ans.at(i) + *iter; 
            }
        }
        
        string answer = "";
        vector<string>::iterator iter1 = ans.begin();
        for(;iter1 < ans.end(); iter1++)
            answer.append(*iter1);
            
        return answer;
    }
};